### PR TITLE
Add installation docs

### PR DESCRIPTION
**Pull request by mattwthompson**
### Description
This PR adds long-overdue installation docs. The docs probably don't render right now, please refer only do the Markdown source.

@umesh-timalsina @justinGilmer I wonder if anything else stuck out to you as pain points when you were installing development builds recently? I wrote up the major sore points that came to mind personally - please let me know what I may be missing.

### Checklist
- [ ] Add tests
- [ ] Lint
- [ ] Update docstrings

### CODE REVIEW

File: docs/installation.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# Installation
+
+These instructions are likely in flux. If something looks out of date or is not working, please raise an issue.
+
+## Install dependencies
+
+Until a conda package is live, the core dependencies are listed in `devtools/conda-envs/minimal_env.yaml` and optional dependencies can be found in `devtools/conda-envs/test_env.yaml`.
```
Comment (umesh-timalsina) [inline docs/installation.md:7]: For me, atleast with the new changes, environment needed also looking at `.github/workflows/ci.yaml`.

File: docs/installation.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  - nbval
+  - mypy
+  - unyt
+  - intermol
+  - gromacs >=2021
+  - lammps
+  - mbuild
+  - foyer>=0.8.0
+
+If portions of the API that require optional dependencies are called while some of those dependencies are not available, an informative error message should be provided. requiring an optional dependency is dependtest d
```
Comment (umesh-timalsina) [inline docs/installation.md:43]: I couldn't infer the remainder of the last sentence.

**Review by umesh-timalsina — COMMENTED** [review]
The installation instructions look good to me and they are in flux anyways.

File: docs/installation.md (hunk 1 above)
Comment (justinGilmer) [inline docs/installation.md:7]: Are only the optional deps here, or is it the _minimal_env.yml_ and the other deps in test_env.yml

**Comment by mattwthompson** [comment]
Thanks for the feedback!

**Pull request merged**